### PR TITLE
Fix requestAnimationFrame in workers

### PR DESCRIPTION
**Pull request by kripken**
Also make the legacy (`mozRequestAnimationFrame` etc.) stuff be behind the `LEGACY_VM_SUPPORT` flag, as it's extremely old at this point (it's even in IE10!).

Update some related worker render tests.

Update setTimeout docs for clarity wrt requestAnimationFrame.

Add a test that verifies we call requestAnimationFrame properly, on the main thread and on workers.

cc @kenrussell

### CODE REVIEW

File: tests/request_animation_frame.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2015 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+// Test RAF is actually used (and not setTimeout etc.)
```
Comment (kenrussell) [inline tests/request_animation_frame.cpp]: It looks like this test verifies that requestAnimationFrame is used for scheduling on the main thread. How does it test that it also works on workers? Is it run by the test harness in multiple modes?

File: tests/test_browser.py [hunk 2]
```diff
@@ -4558,6 +4556,9 @@ def test_emscripten_request_animation_frame(self):
   def test_emscripten_request_animation_frame_loop(self):
     self.btest(path_from_root('tests', 'emscripten_request_animation_frame_loop.c'), '0')
 
+  def test_request_animation_frame(self):
+    self.btest('request_animation_frame.cpp', '0', also_proxied=True)
```
Comment (kenrussell) [inline tests/test_browser.py]: Ah, the also_proxied=True probably answers my question.

Comment (kripken) [reply tests/test_browser.py]: It's a different option, but it's enough to get it running in a worker, which is a good enough test here.

proxy-to-worker basically just runs the entire program in a worker, and does some proxying of mouse and keyboard events, simple 2D canvas display, etc. - enough to run e.g. Doom in a worker. It doesn't use pthreads or anything like that. We may eventually reimplement it using pthreads.

**Comment by kripken** [comment]
Thanks for the comments!

**Comment by kripken** [comment]
cc @nickshinpho - this PR may be useful to cherry-pick, but I don't think it's strictly necessary (as rendering should work even without requestAnimationFrame, just not as well).

**Pull request merged**